storage/store/headers.go

```diff
@@ -188,7 +187,7 @@ func (h *Headers) BlockIDByHeight(height uint64) (flow.Identifier, error) {
 // CAUTION: this method is not backed by a cache and therefore comparatively slow!
 func (h *Headers) ByParentID(parentID flow.Identifier) ([]*flow.Header, error) {
```

Comment:
This function is missing error returns. Please add them. 

You should view this comment as a follow-up to [my previous comment](url). 
A dominant pattern in our storage abstractions is the following
https://github.com/onflow/flow-go/blob/fc77660d737e7c5b8f146d7ea6e76b31f79f5738/storage/store/headers.go#L149

We return `storage.ErrNotFound` if the entity about which we request information is not known. Your API change breaks with this convention. [`ByParentID`](https://github.com/onflow/flow-go/blob/fc77660d737e7c5b8f146d7ea6e76b31f79f5738/storage/store/headers.go#L188) is now a function that subtly behaves differently. 

```suggestion
//
// Expected error returns during normal operations:
//   - [storage.ErrNotFound] if no block with the given parentID is known
func (h *Headers) ByParentID(parentID flow.Identifier) ([]*flow.Header, error) {
```